src/com/amazon/corretto/crypto/provider/LibCryptoRng.java

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.corretto.crypto.provider;
+
+import java.security.SecureRandom;
+import java.security.SecureRandomSpi;
+
+/**
+ * A simple wrapper around the linked LibCrypto's RAND_bytes() API.
+ */
+public class LibCryptoRng extends SecureRandom {
+    private static final String ALGORITHM_NAME = "LibCryptoRng";
+    private static final long serialVersionUID = 1L;
+    private static final int MAX_SINGLE_REQUEST = 8192;
+
+    private static native void generate(byte[] bytes, int offset, int length);
+
+    public LibCryptoRng() {
+        super(new SPI(), AmazonCorrettoCryptoProvider.INSTANCE);
+    }
+
+    @Override
+    public String getAlgorithm() {
+        return ALGORITHM_NAME;
+    }
+
+    static class SPI extends SecureRandomSpi {
+        private static final long serialVersionUID = 1L;
+        static final SelfTestSuite.SelfTest SELF_TEST = new SelfTestSuite.SelfTest(ALGORITHM_NAME, LibCryptoRng.SPI::runSelfTest);
+
+        static SelfTestResult runSelfTest() {
+            LibCryptoRng rnd = new LibCryptoRng();
+
+            /* Basic RNG self-test, do we generate different random numbers on repeat calls? */
+            final long initialLong = rnd.nextLong();
+            for (int trial = 0; trial < 10; trial++) {
+                if (initialLong != rnd.nextLong()) {
+                    return new SelfTestResult(SelfTestStatus.PASSED);
+                }
+            }
+            return new SelfTestResult(SelfTestStatus.FAILED);
+        }
+
+        SPI() {
+        }
+
+        @Override
+        protected byte[] engineGenerateSeed(int numBytes) {
+            final byte[] seed = new byte[numBytes];
+            engineNextBytes(seed);
+            return seed;
+        }
+
+        @Override
+        protected void engineNextBytes(byte[] bytes) {
+            int offset = 0;
+            while (offset < bytes.length) {
+                final int toGenerate = Math.min(MAX_SINGLE_REQUEST, bytes.length - offset);
```

Comment:
This value was taken from [the previous AesCtrDrbg implementation](https://github.com/corretto/amazon-corretto-crypto-provider/blob/80b3fa918a4787dd5d1d4e3b2451f7e0e5ae5aad/src/com/amazon/corretto/crypto/provider/AesCtrDrbg.java#L22), which I think was taken from [page 49 of the NIST SP800-90A document](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf#page=58). I kept the 8k limit to be on the safe side, but it may not be needed since this RNG doesn't internally use AES anymore.